### ISSUE TITLE
Handle Error of SharedMemError::LinkExists then link is removed

### ISSUE THREAD

**Issue by owen-hope** [elast0ny/mem_file#40]
Hi I am trying to figure out how to handle a situation where in one process a shared memory link would be created then go out of scope with nothing being written to it. Which would then cause it to be removed from memory. 

In parallel another process would be creating a shared memory link. It would see the link existed and fall into the caught error of `Err(SharedMemError::LinkExists) => SharedMem::open_linked(link_path)?,` At this point the first process has already removed the shared memory so when process two goes to open the link with that specified link path it is no longer there. This leads to the error of `Error: MapOpenFailed(22)`.

Is there an easy solution to this problem?

I have tried to do something like this after the first match config.create() 
`Err(SharedMemError::LinkExists) => {
  match SharedMem::open_linked(link_path) {
    Ok(x) => x,
     Err(SharedMemError::MapOpenFailed(22)) => {
       config.create()?
      },
  }
}`

The issue with this is that SharedMemConf doesn't implement the Copy trait. Which is there a specific reason as for why it cannot implement copy?

**Comment by elast0ny** [comment]
Hi, i'm not sure I understand the situation you are explaining but I tried adding `#[derive(Clone, Copy)]` on `SharedMemConf` but unfortunately, it contains a struct [theban_interval_tree::IntervalTree](https://docs.rs/theban_interval_tree/0.7.1/theban_interval_tree/tree/struct.IntervalTree.html) which does not implement Clone/Copy... This struct is used to make sure lock ranges don't overlap. If another interval tree type struct exists out there that implements Clone/Copy, I wouldn't be opposed to switching to it.

If what you're describing is : 
1. `Proc1` creates a mapping with link file `link123`
2. `Proc2` attempts to create/open with link file `link123`, detects that it already exists
3. `Proc1` Closes the mapping and because it created it, it deletes `link123`
4. `Proc2` attempts to open `link123`

Then you could run the create/open logic of `Proc2` in a loop until it succeeds. I would imagine this race condition shouldn't happen often unless you are creating/deleting mappings extremely frequently in multiple processes.

**Comment by elast0ny** [comment]
Will close this unless there is anything else to address
